azure-mgmt-samples/pom.xml

```diff
@@ -68,6 +68,11 @@
         </dependency>
         <dependency>
             <groupId>${project.groupId}</groupId>
+            <artifactId>azure-core-test</artifactId>
```

Comment:
It shouldn't. This package only contains one class `MockIntegrationTestBase` which is used by tests to do the recording/mocking. I don't see it being used in any of your changes. 
A cleanup or re-import in IntelliJ might solve the problem. If not, `mvn clean compile` should always tell the truth.